### PR TITLE
Added some generics and removed usages of deprecated methods

### PR DESCRIPTION
**Pull request by tonihele**

### CODE REVIEW

File: jme3-templates/src/com/jme3/gde/templates/basic/BasicGamePanelVisual.java [hunk 1]
```diff
@@ -110,7 +110,7 @@ private void browseButtonActionPerformed(java.awt.event.ActionEvent evt) {//GEN-
         String command = evt.getActionCommand();
         if ("BROWSE".equals(command)) {
             JFileChooser chooser = new JFileChooser();
-            FileUtil.preventFileChooserSymlinkTraversal(chooser, null);
```
Comment (MeFisto94) [inline jme3-templates/src/com/jme3/gde/templates/basic/BasicGamePanelVisual.java]: I am not sure about that, why has it been removed and what could be a replacement?

File: jme3-tests-template/src/com/jme3/gde/templates/tests/testsandroid/JME3TestsAndroidPanelVisual.java [hunk 2]
```diff
@@ -110,7 +110,7 @@ private void browseButtonActionPerformed(java.awt.event.ActionEvent evt) {//GEN-
         String command = evt.getActionCommand();
         if ("BROWSE".equals(command)) {
             JFileChooser chooser = new JFileChooser();
-            FileUtil.preventFileChooserSymlinkTraversal(chooser, null);
```
Comment (MeFisto94) [inline jme3-tests-template/src/com/jme3/gde/templates/tests/testsandroid/JME3TestsAndroidPanelVisual.java]: Same here. Does setting the current directory to null prevent symlinks (Note: I didn not read the javadoc, I don't know what the methods do)

File: jme3-tests-template/src/com/jme3/gde/templates/tests/testsdesktop/JmeTestsPanelVisual.java [hunk 3]
```diff
@@ -110,7 +110,7 @@ private void browseButtonActionPerformed(java.awt.event.ActionEvent evt) {//GEN-
         String command = evt.getActionCommand();
         if ("BROWSE".equals(command)) {
             JFileChooser chooser = new JFileChooser();
-            FileUtil.preventFileChooserSymlinkTraversal(chooser, null);
+            chooser.setCurrentDirectory(null);
```
Comment (MeFisto94) [inline jme3-tests-template/src/com/jme3/gde/templates/tests/testsdesktop/JmeTestsPanelVisual.java]: and here

**Comment by tonihele** [comment]
http://bits.netbeans.org/dev/javadoc/org-openide-filesystems/org/openide/filesystems/FileUtil.html

Apparently that method was there to fix a bug that existed in Java 5 and
earlier. The link says to use the other method instead. Im not sure if it
is needed or not, it might be the default behavior anyway. For me it opens
to My documents in Windows.

**Pull request merged**